src/main/scala/com/danielasfregola/twitter4s/TwitterRestClient.scala

```diff
@@ -62,23 +63,50 @@ trait RestClients
     with TwitterUserClient
 
 object TwitterRestClient {
-
   def apply(): TwitterRestClient = {
-    val consumerToken = ConsumerToken(key = consumerTokenKey, secret = consumerTokenSecret)
-    val accessToken = AccessToken(key = accessTokenKey, secret = accessTokenSecret)
-    apply(consumerToken, accessToken)
+    // See what we have to work with
+    authMode match {
+      case UseOAuth1 => {
+        val consumerToken = ConsumerToken(key = consumerTokenKey.get, secret = consumerTokenSecret.get)
```

Comment:
Using `.get` when using options is bad because it will throw an exception when the optional value is empty